### ISSUE TITLE
Bring in the improvements branch

### ISSUE THREAD

**Issue by grounded042** [Chatham/LetsTrace#5]
@Falco20019 has an improvements branch that we have both been working on (mostly him :)).

It would be nice to get this merged in an incorporated soon. I'm creating this so we can track what all needs to be done before we merge.

**Comment by grounded042** [comment]
@Falco20019 do you know what all needs to be done before you're ready to merge?

**Comment by Falco20019** [comment]
I think we only need to get the tests done. Functionality should be complete and on par with `jaeger-client-java`. A code review would also be nice, just to ensure maintainability. Right now, I'm filling in the missing tests based on coverage analysis.

I also would love to have thrift excluded and in an official NuGET package, but that only affects `LetsTrace.Jaeger`.

**Comment by grounded042** [comment]
I'm with you on the Thrift issue - we *could* publish the package ourself, but I really don't want to do that.

I'm working through tests - right now we are at 80% coverage and I'd like to get to at least 90%. As for a code review I'm going to have some other guys from Chatham take a look, but others are welcome.

We are starting to look at getting this into production, so we'll also be able to see how using it actually works.

**Comment by grounded042** [comment]
My goal is to get this merged in by the end of the week.

**Comment by Falco20019** [comment]
@mweinand and @yurishkuro offered help, so maybe they could look into doing a code review. I'm out of office for the rest of this week, so I only will try to get tests and constants done the next couple hours. I assume, the rest of the TODOs can be postponed since they haven't been part of LetsTrace before and are also not mandatory according to https://github.com/jaegertracing/jaeger/issues/366

**Comment by grounded042** [comment]
I need to go through and change a lot of the TODOs to issues here on GitHub

**Comment by grounded042** [comment]
I'll look at getting things reviewed and merged by the end of the week and maybe when you come back we'll have a new package ready to go!

**Comment by Falco20019** [comment]
Sounds like a plan. I'm pretty happy with the improvements and how quickly we got them done so far. Thought it would take longer to implement :) Thanks for the support on the unit tests, greatly appreciate it.

**Comment by grounded042** [comment]
Yeah, no problem. Glad we could join forces on this project. Can't wait to see where it goes.

**Comment by Falco20019** [comment]
We are at 95% coverage now. `PerOperationSampling` needs some more love but I'm out of office now. Maybe someone could extend the `RemoteControlledSamplerTest`, which should cover nearly all of the uncover test surface. An `TODO` notice is left in the code.

**Comment by yurishkuro** [comment]
you guys rock!

**Comment by grounded042** [comment]
@Falco20019 sweet! I've actually been working on the `PerOperationSampling` tests today, so we're all good!

**Comment by grounded042** [comment]
seeing if we could replace `NullReporter` with `LoggingReporter(NullLoggerFactory.Instance)` was mentioned above, but I'm going to leave the `NullReporter` in there as Java also has one. I like explicitly defining a null reporter.

**Comment by grounded042** [comment]
We have tests done with 99% coverage right now. I'm going to have some guys here at Chatham do the review to merge it in. Once that is complete we'll working on making sure LetsTrace.Jaeger merges in the improvements branch. Once that is done we can merge the two together and get working on moving everything into the jaegertracing org.

**Comment by mweinand** [comment]
Nice work!

**Comment by grounded042** [comment]
This has been merged. I bumped the version and updated the README as well. Next step is to get the improvements branch merged into LetsTrace.Jaeger and then merge that repo with this. Before we execute the move to the Jaeger org we'll probably want to put some compiler warnings in for anyone using the lib under it's old name.
